api/routes/books.js

```diff
@@ -46,7 +55,12 @@ router.post('/', async (req, res, next) => {
 router.patch('/:id/reserve', async (req, res, next) => {
   const { id } = req.params
   try {
+    if (req.headers.authorization === null) {
```

Comment:
This doesn't actually check to see whether or not someone is logged in, it only checks to see whether or not someone has sent over a token. If the token is invalid, it will pass with the given code.